### PR TITLE
pam_nologin: don't print empty message

### PR DESCRIPTION
**Pull request by thkukuk**
Conversation functions most of the time add an extra "\n" to messages from PAM modules. This leads to disturbing line breaks and empty lines in the output.
Better to do not print an empty message at all.

### CODE REVIEW

File: modules/pam_nologin/pam_nologin.c [hunk 1]
```diff
@@ -99,21 +99,25 @@ static int perform_check(pam_handle_t *pamh, struct opt_s *opts)
...
-
-	if (pam_modutil_read(fd, mtmp, st.st_size) == st.st_size) {
+	/* Don't print anything if the message is empty, will only
+	   disturb the output with empty lines */
+	if (st.st_size > 0) {
+	    mtmp = malloc(st.st_size+1);
+	    if (!mtmp) {
+	        pam_syslog(pamh, LOG_CRIT, "out of memory");
+		retval = PAM_BUF_ERR;
+		goto clean_up_fd;
```
Comment (ikerexxe) [inline modules/pam_nologin/pam_nologin.c:109]: Different indentation level for content inside a block.

File: modules/pam_nologin/pam_nologin.c [hunk 2]
```diff
@@ -99,21 +99,25 @@ static int perform_check(pam_handle_t *pamh, struct opt_s *opts)
...
+	    }
+
+	    if (pam_modutil_read(fd, mtmp, st.st_size) == st.st_size) {
 		mtmp[st.st_size] = '\0';
 		(void) pam_prompt (pamh, msg_style, NULL, "%s", mtmp);
```
Comment (ikerexxe) [inline modules/pam_nologin/pam_nologin.c:114]: The content inside the block appears to have the same level of indentation as that outside the block.

File: modules/pam_nologin/pam_nologin.c (hunk 1 above)
Comment (thkukuk) [reply modules/pam_nologin/pam_nologin.c:109]: Github does not seem to show TABs and whitespaces correct...

File: modules/pam_nologin/pam_nologin.c (hunk 2 above)
Comment (thkukuk) [reply modules/pam_nologin/pam_nologin.c]: Don't trust your webbrowser...

File: modules/pam_nologin/pam_nologin.c [hunk 3]
```diff
@@ -99,21 +99,25 @@ static int perform_check(pam_handle_t *pamh, struct opt_s *opts)
...
-	    goto clean_up_fd;
-	}
-
-	if (pam_modutil_read(fd, mtmp, st.st_size) == st.st_size) {
-		mtmp[st.st_size] = '\0';
-		(void) pam_prompt (pamh, msg_style, NULL, "%s", mtmp);
+	/* Don't print anything if the message is empty, will only
+	   disturb the output with empty lines */
+	if (st.st_size > 0) {
+	    mtmp = malloc(st.st_size+1);
```
Comment (ldv-alt) [inline modules/pam_nologin/pam_nologin.c:105]: Could you also move the declaration of `mtmp` from
https://github.com/linux-pam/linux-pam/blob/016d5e2bc0423f5226191143329ad7e50fde82fd/modules/pam_nologin/pam_nologin.c#L82
inside this new block, e.g.
```
if (st.st_size > 0) {
    char *mtmp = malloc(st.st_size+1);
```
?

**Review by ldv-alt — APPROVED** [review]

**Pull request merged**